server/src/projects/projects.service.ts

```diff
@@ -229,7 +229,15 @@ export class ProjectsService {
 
           ...projectMilestone,
         }));
-      const formattedProjectMilestones = overwriteCodesWithLabels(projectMilestones, [...MILESTONE_ATTRS]);
+
+      // if we include dates when we reformat below then the dates end up being sent to the frontend as strings,
+      // so we exclude them here
+      const MILESTONE_ATTRS_WITHOUT_DATES = [
```

Comment:
optional suggestion:
Maybe instead of defining this here, in `projects/projects.attrs.ts` , you can compose MILESTONE_ATTRS using two different subsets, like 
```
export const MILESTONE_DATE_ATTRS = [
  'dcp_actualstartdate',
  'dcp_actualenddate',
  'dcp_plannedcompletiondate',
  'dcp_plannedstartdate',
];

export const MILESTONE_CORE_ATTRS = [ // or MILESTONE_NON_DATE_ATTRS?
 'statuscode',
 '_dcp_milestone_value',
]

export const MILESTONE_ATTRS = [
   ...MILESTONE_DATE_ATTRS,
   ...MILESTONE_CORE_ATTRS,
];
```

Then import the MILESTONE_CORE_ATTRS array in this file.
```
import { MILESTONE_ATTRS, MILESTONE_CORE_ATTRS } from './projects.attrs';
```

Reply:
Good idea! I'll make those changes